### PR TITLE
prompt user to save file when 'Content-Disposition' tag was set in th…

### PR DESCRIPTION
**Pull request by zenglanmu**
when the server side response with Content-Disposition tag, prompt user t save file with the filename came from filename section in the Content-Disposition tag. And set body content as "body not available; it has been saved as a file because of the content-disposition header"

### CODE REVIEW

**Review by frigus02 — COMMENTED** [review]
I'm happy with the approach. Thanks for working on this. Just a few comments.

File: src/site/elements/data/scripts/request.js [hunk 1]
```diff
@@ -360,9 +375,24 @@ export async function send(request) {
         );
     }
 
-    const fetchBody = await fetchResponse.text();
     response.timeEnd = new Date();
```
[the remark this answers is not in the document]
Comment (zenglanmu) [reply src/site/elements/data/scripts/request.js]: add response.timeEnd = new Date() after each case

**Comment by frigus02** [comment]
This looks really nice now, I think.

>  And not sure we should create download url from response.blob(), or pass the request.url and header directly to chrome.downloads.download.

Generating a URL from response.blob() is probably a good start. This ensures that we only send the request once, which might also be important. If can tackle the additional memory allocation if and when it becomes a problem.

Remaining steps for this PR are:

- Clean up lint errors. You can reproduce them locally by running `yarn lint`

- I would love to have some unit tests for `getFilenameFromContentDispositionHeader`. That would help me and future readers to understand how the header might look like. You can export the function, create a file `request.test.js` and add some tests along the lines of:

  ```js
  describe('getFilenameFromContentDispositionHeader', function () {
      test('utf8', function () {
          expect(getFilenameFromContentDispositionHeader('attachment; filename*=foo')).toEqual('foo');
      });
      test('ascii', function () { ... });
      test('reverved characters', function () { ... });
      ...
  });
  ```

  Once the file exists you should be able to run them using `yarn test` and optionally `yarn test --watch`

**Comment by zenglanmu** [comment]
> 

lint fixed and test for function getFilenameFromContentDispositionHeader added

**Comment by frigus02** [comment]
Sorry for the delay, @zenglanmu. I think you were correct originally to revoke the object url after the download is complete. I added it back and did the same in another place where I forgot to do it earlier. Also ran the formatter across everything.

I'm going to merge this when the build passes and then publish a new version of RESTer.

**Pull request merged**